PR: https://github.com/odoo/odoo/pull/245827

From: 3cce5fbeba01e3b1aa453c13c5d973826a4f0a2a
From: ksja-odoo
Date: 2026-05-05 06:17:53

Breaking data model changes scores: del:4 + add:0, change matches:
-    stage_id_color = fields.Integer(string='Stage Color', related="stage_id.color", export_string_translation=False)
-    color = fields.Integer(string='Color', export_string_translation=False)
-    stage_id_color = fields.Integer(string='Stage Color', related="stage_id.color", export_string_translation=False)
-    color = fields.Integer(string='Color', export_string_translation=False)

Total Changes: 5

[IMP] project, crm: remove the color field from stages

We used the rotting widget in the stage field, so the stage
color field is unused, as explained below in the commit.
https://github.com/odoo/odoo/commit/cd9ed578803605550135cb6fd6d4c267b433abbc

task-5485507

closes odoo/odoo#245827

Related: odoo/enterprise#105595
Related: odoo/upgrade#9575
Signed-off-by: Xavier Bol (xbo) <xbo@odoo.com>

================================= pseudo patch: =================================

--- a/addons/project/models/project_project.py
+++ b/addons/project/models/project_project.py
@@ -162,7 +162,6 @@ class ProjectProject(models.Model):
     # Not `required` since this is an option to enable in project settings.
     stage_id = fields.Many2one('project.project.stage', string='Stage', ondelete='restrict', groups="project.group_project_stages",
         tracking=True, index=True, copy=False, default=_default_stage_id, group_expand='_read_group_expand_full')
-    stage_id_color = fields.Integer(string='Stage Color', related="stage_id.color", export_string_translation=False)
     duration_tracking = fields.Json(groups="project.group_project_stages")
     rotting_days = fields.Integer(groups="project.group_project_stages")
     is_rotting = fields.Boolean(groups="project.group_project_stages")

--- a/addons/project/models/project_project_stage.py
+++ b/addons/project/models/project_project_stage.py
@@ -18,7 +18,6 @@ class ProjectProjectStage(models.Model):
     fold = fields.Boolean('Folded',
         help="If enabled, this stage will be displayed as folded in the Kanban and List views of your projects. Projects in a folded stage are considered as closed.")
     company_id = fields.Many2one('res.company', string="Company")
-    color = fields.Integer(string='Color', export_string_translation=False)
     rotting_threshold_days = fields.Integer('Days to rot', default=0, help='Day count before projects in this stage become stale. Set to 0 to disable.')
 
     def copy_data(self, default=None):

--- a/addons/project/models/project_task.py
+++ b/addons/project/models/project_task.py
@@ -60,7 +60,6 @@ PROJECT_TASK_READABLE_FIELDS = {
     'is_template',
     'has_template_ancestor',
     'has_project_template',
-    'stage_id_color',
     'access_token',
     'access_url',
 }
@@ -164,7 +163,6 @@ class ProjectTask(models.Model):
        store=True, readonly=False, ondelete='restrict', tracking=True, index=True,
        default=_get_default_stage_id, group_expand='_read_group_stage_ids',
        domain="[('project_ids', '=', project_id)]")
-    stage_id_color = fields.Integer(string='Stage Color', related="stage_id.color", export_string_translation=False)
     tag_ids = fields.Many2many('project.tags', string='Tags')
 
     state = fields.Selection([

--- a/addons/project/models/project_task_type.py
+++ b/addons/project/models/project_task_type.py
@@ -32,7 +32,6 @@ class ProjectTaskType(models.Model):
         string='Email Template',
         domain=[('model', '=', 'project.task')],
         help="If set, an email will be automatically sent to the customer when the task reaches this stage.")
-    color = fields.Integer(string='Color', export_string_translation=False)
     fold = fields.Boolean(string='Folded')
     rating_template_id = fields.Many2one(
         'mail.template',
